src/frequenz/sdk/timeseries/_periodic_feature_extractor.py

```diff
@@ -0,0 +1,414 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+A class that extracts features out of historical data.
```

Comment:
I would remove the "a class that", maybe something like:

```suggestion
An extractor to get features out of historical data.
```

Also, please bear in mind that this module is **private**, so all this documentation won't be shown in the rendered documentation in the website. I'd suggest moving all the important documentation to the class instead.